clang/include/clang/CIR/Dialect/IR/CIRAttrs.td

```diff
@@ -708,6 +708,23 @@ def SignedOverflowBehaviorAttr : AttrDef<CIR_Dialect, "SignedOverflowBehavior">
   let hasCustomAssemblyFormat = 1;
 }
 
+//===----------------------------------------------------------------------===//
+// ByteOffsetAttr
+//===----------------------------------------------------------------------===//
+
+def ByteOffsetAttr : CIR_Attr<"ByteOffset", "byte_offset"> {
```

Comment:
Instead of adding `ByteOffsetAttr`, shouldn't we just use mlir::IntegerAttr directly?

Reply:
we can! I just thought it would be more explicit